src/lib.rs

```diff
@@ -184,32 +187,35 @@ pub trait MysqlShim<W: Write> {
     fn on_init(&mut self, _: &str, _: InitWriter<'_, W>) -> Result<(), Self::Error> {
         Ok(())
     }
+
+    /// Provides the TLS configuration, if we want to support TLS.
+    fn tls_config(&self) -> Option<&rustls::ServerConfig> {
```

Comment:
Since what we ultimately need is `Arc<rustls::ServerConfig>`, let's have the caller directly provide that. That way, they can re-use a single `Arc` rather than us creating a new one for each connection. It _also_ allows them to create configs on-the-fly, which the current configuration does not.

Reply:
Fixed.